planner/core/exhaust_physical_plans.go

```diff
@@ -1483,16 +1483,40 @@ func (p *LogicalWindow) exhaustPhysicalPlans(prop *property.PhysicalProperty) []
 	var byItems []property.Item
 	byItems = append(byItems, p.PartitionBy...)
 	byItems = append(byItems, p.OrderBy...)
+
+	concurrency := p.ctx.GetSessionVars().WindowConcurrency
+	if concurrency > 1 { //TODO: implement a cost-based strategy
+		return p.getWindowParallel(prop, byItems)
+	}
+	return p.getWindow(prop, byItems)
+}
+
+func (p *LogicalWindow) getWindow(prop *property.PhysicalProperty, byItems []property.Item) []PhysicalPlan {
 	childProperty := &property.PhysicalProperty{ExpectedCnt: math.MaxFloat64, Items: byItems, Enforced: true}
 	if !prop.IsPrefix(childProperty) {
 		return nil
 	}
-	window := PhysicalWindow{
+	window := PhysicalWindow{BasePhysicalWindow{
+		WindowFuncDescs: p.WindowFuncDescs,
+		PartitionBy:     p.PartitionBy,
+		OrderBy:         p.OrderBy,
+		Frame:           p.Frame,
+	}}.Init(p.ctx, p.stats.ScaleByExpectCnt(prop.ExpectedCnt), p.blockOffset, childProperty)
+	window.SetSchema(p.Schema())
+	return []PhysicalPlan{window}
+}
+
+func (p *LogicalWindow) getWindowParallel(prop *property.PhysicalProperty, byItems []property.Item) []PhysicalPlan {
+	if !prop.IsEmpty() {
+		return nil
+	}
```

Comment:
Should not `return nil` or it will have no plan. We may judge `!prop.IsEmpty()` in `exhaustPhysicalPlans`.